power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -171,13 +164,29 @@ template <bool sym> class NewtonRaphsonSESolver {
     // data for gain matrix
     std::vector<NRSEGainBlock<sym>> data_gain_;
     // unknown and rhs
-    std::vector<NRSERhs<sym>> del_x_rhs_;
+    std::vector<NRSERhs<sym>> delta_x_rhs_;
     // voltage of current iteration
     std::vector<NRSERhs<sym>> x_;
     // solver
     SparseLUSolver<NRSEGainBlock<sym>, NRSERhs<sym>, NRSEUnknown<sym>> sparse_solver_;
     typename SparseLUSolver<NRSEGainBlock<sym>, NRSERhs<sym>, NRSEUnknown<sym>>::BlockPermArray perm_;
 
+    // array selection function pointer
+    static constexpr std::array has_branch_{&MeasuredValues<sym>::has_branch_from, &MeasuredValues<sym>::has_branch_to};
+    static constexpr std::array branch_power_{&MeasuredValues<sym>::branch_from_power,
+                                              &MeasuredValues<sym>::branch_to_power};
+
+    void initialize_unknown(MathOutput<sym>& output, MeasuredValues<sym> const& measured_values) {
+        RealValue<sym> const mean_angle_shift = measured_values.mean_angle_shift();
+        for (Idx bus = 0; bus != n_bus_; ++bus) {
+            x_[bus].v() = 1.0;
```

Comment:
According to the math workout:
For bus i:
- if no voltage measurement: `U_i = 1`, `theta_i = e^j*theta_i` (what you are doing here)
- If voltage measurement `U_i = U_meas_i`, `theta_i = e^j*theta_i`; Is this done in a later stage or is this something missing in this function?

Reply:
yes, nitish found it and it's being resolved now